src/actions/importer.js

```diff
@@ -139,23 +144,39 @@ export function executeImport(tableId) {
 
         // Find the gender field if it exists
         let genderIdx = columns.findIndex(i => i.type == 'person_gender');
+        // Find any json custom fields
+        let jsonFieldIdxs = columns.reduce((arr, col, idx) => {
+            if(col.type == 'person_field' && col.config.field_type == 'json') {
+                arr.push(idx);
+            }
+            return arr;
+        }, []);
 
-        // Map gender column
-        if(genderIdx > -1) {
-            let mappings = columns[genderIdx].config.mappings;
+        if(genderIdx > -1 || jsonFieldIdxs.length) {
+            let genderMapping = genderIdx > -1 ? columns[genderIdx].config.mappings : null;
             rows = rows.map(r => {
-                let value = r[genderIdx];
-                let new_value = mappings.find(m => {
-                    if(m.value === value) {
-                        return true;
-                    }
-                });
-                r[genderIdx] = new_value && new_value.gender ? new_value.gender : null;
+                // Map gender column
+                if(genderIdx > -1) {
+                    let value = r[genderIdx];
+                    let new_value = genderMapping.find(m => {
+                        if(m.value === value) {
+                            return true;
+                        }
+                    });
+                    r[genderIdx] = new_value && new_value.gender ? new_value.gender : null;
+                }
+                // Parse JSON column from text to structured JSON
+                for(const jsonIdx of jsonFieldIdxs) {
+                    let value = r[jsonIdx];
+                    r[jsonIdx] = JSON.parse(value);
```

Comment:
This is not dealing correctly with empty cells. I tried to find an easy solution (e.g. `!!value? JSON.parse(value) : null`) but I think that only introduces other issues, like how do we deal with accidental overwriting when cells are empty? There also seems to be some issues with how arrays are handled when sent off to the back-end.

My suggestion is that we don't support JSON fields at all in the importer. I.e. filter them out in `ImporterColumnHead` and remove the validation/parsing code.